src/Data/Refinement.agda

```diff
@@ -0,0 +1,45 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Refinement type: a value together with an erased proof.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Refinement where
```

Comment:
Again, should this be under `Data`? (I'm not sure, just putting the thought out there!)

Reply:
I see it as a `Data.Product` where the second component is erased.
But I guess it could also be seen as a new quantifier in `Relation.Unary`.